File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* mergeInBetween(ListNode* head, int a, int b, ListNode* head2) {
    int n=0;
    ListNode* temp=head;
    while(temp!=NULL){
        temp=temp->next;
        n++;
    } 
    // marak all position
    ListNode* first=head;
    for(int i=1;i<a;i++){
        first=first->next;
    }

    ListNode* sec=head;
    for(int i=1;i<b+2;i++){
         sec=sec->next;
    } 
    first->next=head2;
    ListNode* temp2=head2;

    while(temp2->next!=NULL){
        temp2=temp2->next;
    }
    temp2->next=sec;
    return head;
    }
};

File: 1765-merge-in-between-linked-lists/merge-in-between-linked-lists.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode mergeInBetween(ListNode head1, int a, int b, ListNode head2) {
        ListNode first = head1;
        ListNode second = head1;

        // Comment dal diye hai 

        
        // Move first to node just before index a
        for (int i = 0; i < a - 1; i++) {
            first = first.next;
        }

        // Move second to node at index b
        for (int i = 0; i < b; i++) {
            second = second.next;
        }

        // Connect first.next to head2
        first.next = head2;

        // Find tail of head2
        ListNode tail = head2;
        while (tail.next != null) {
            tail = tail.next;
        }

        // Connect tail.next to second.next
        tail.next = second.next;

        return head1;
    }
}
